docs/DESKTOP_ENVIRONMENTS.md

```diff
@@ -0,0 +1,118 @@
+---
+title: Desktop Environment integration
+category: Concepts
+layout: default
+---
+
+# Desktop Environments
+
+NOTE: This document is work-in-progress.
+
+## Single Graphical Session
+
+To be able to manage a graphical session using systemd, there is a strong
+requirement that each user only has one such session running at a time. While
+this might not have always been the case historically, having multiple sessions
+for one user running at the same time is problematic for a number of reasons
+(including e.g. the DBus session bus being shared).
+
+Note that in principle it is possible to run a single graphical session
+across multiple logind seats in principle. i.e. when the user logs in to a
+second seat, one could assign the seats resources to the existing session,
+effectively presenting it as a single seat.
+
+## Pre-defined systemd units
+
+[`systemd.special(7)`](https://www.freedesktop.org/software/systemd/man/systemd.special.html)
+defines the `graphical-session.target` and `graphical-session-pre.target` to
+allow cross-desktop integration. Furthermore, systemd defines the three base
+slices `background`, `apps` and `session`. All units should be placed into
+one of these slices depending on their purposes:
+
+ * `session.slice`: Contains only processes essential to run the users graphical session
+ * `apps.slice`: Contains all normal applications that the user is running
+ * `background.slice`: Useful for low-priority background tasks
+
+The purpose of this grouping is to assign different priorities to the
+applications. This could e.g. mean reserving memory to session processes,
+preferentially killing background tasks in out-of-memory situations
+or assinging different memory/CPU/IO priorities to ensure that the session
+runs smoothly under load.
+
+TODO: Will there be a default to place units into e.g. `apps.slice` by default
+rather than the root slice?
+
+## XDG standardization for applications
+
+To ensure cross-desktop compatibility and encourage sharing of good practices,
+desktop environments should adhere to the following conventions:
+
+ * Application units should follow the scheme `apps-launcher-ApplicationID-PID.service`,
```

Comment:
Maybe use `<>`: `apps-<launcher>-<ApplicationID>-<PID>.service` ?

It should be mentioned what `<PID>` is. Because it cannot be the PID of the process, since it's not known when the unit is started.

Reply:
We currently use a UUID.

One reason for that being that if something in a PID namespace tries to launch something we could create a clash.

Reply:
It would likely have to be some kind of UUID instead

Reply:
Yeah, I meant to say <RAND> or rather than PID.

I am not convinced about using a UUID. Not because I dislike using a UUID in general. But because they contain the "-" character and it seems sensible to me to avoid it as it has special meanings to systemd.